### PR TITLE
Update docs to use relative code links.

### PR DESCRIPTION
**Pull request by jpeach**
## Changes

Use relative links to locations within the same repository. This means
that in the Github UI the reader will follow the link to the location
in the same branch rather than to a fixed, predetermined branch.

### CODE REVIEW

File: docs/dependencies.md [hunk 1]
```diff
@@ -10,19 +10,18 @@ Both these dependencies are listed here:
 
 ## Internal dependencies
 
-- [nostd::variant](https://github.com/open-telemetry/opentelemetry-cpp/blob/v1.0.0-rc1/api/include/opentelemetry/nostd/absl/types/variant.h):
+- [nostd::variant](../api/include/opentelemetry/nostd/absl/types/variant.h):
```
Comment (lalitb) [inline docs/dependencies.md:13]:
```suggestion
- [nostd::variant](../api/include/opentelemetry/nostd/internal/absl/types/variant.h):
```

**Review by lalitb — APPROVED** [review]
Thanks. LGTM with a minor fix.

File: docs/dependencies.md (hunk 1 above)
Comment (jpeach) [reply docs/dependencies.md]: Thanks, applied!

**Comment by jpeach** [comment]
docfx hates this apparently

```
[22-06-11 07:14:47.612]Warning:[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](docs/dependencies.md#L17)Invalid file link:(~/exporters/etw/include/opentelemetry/exporters/etw/TraceLoggingDynamic.h).
[22-06-11 07:14:47.612]Warning:[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](docs/dependencies.md#L24)Invalid file link:(~/sdk).
```

**Comment by lalitb** [comment]
> docfx hates this apparently
> 
> ```
> [22-06-11 07:14:47.612]Warning:[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](docs/dependencies.md#L17)Invalid file link:(~/exporters/etw/include/opentelemetry/exporters/etw/TraceLoggingDynamic.h).
> [22-06-11 07:14:47.612]Warning:[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](docs/dependencies.md#L24)Invalid file link:(~/sdk).
> ```

Yes, I think we can fix `TraceLoggingDynamic.h` error by adding the `exporters/**.{cc|h}` in the `resource/files` scope in `docfx.json`, but we need to check how to support relative links to internal github directories. Let me check if I can find something. Tagging @reyang just in case he has an idea.

**Comment by ThomsonTan** [comment]
Could we please try remove the prefix `..` and make it path relative to the repo?

**Comment by jpeach** [comment]
Updated to use root-relative links, and docfx seems happy.

**Pull request merged**